domain-models-runtime/src/main/java/org/folio/rest/persist/PostgresClient.java

```diff
@@ -2379,26 +2379,42 @@ private FacetManager buildFacetManager(CQLWrapper wrapper, QueryHelper queryHelp
 
   /**
    * Get the jsonb by id.
+   * @param conn  if provided, the connection on which to execute the query on.
+   * @param lock  whether to use SELECT FOR UPDATE to lock the selected row.
    * @param table  the table to search in
    * @param id  the value of the id field
    * @param function  how to convert the (String encoded) JSON
    * @param replyHandler  the result after applying function
    */
-  private <R> void getById(String table, String id,
+  private <R> void getById(final AsyncResult<SQLConnection> conn, boolean lock, String table, String id,
                            FunctionWithException<String, R, Exception> function,
                            Handler<AsyncResult<R>> replyHandler) {
-    getConnection(res -> {
-      if (res.failed()) {
-        replyHandler.handle(Future.failedFuture(res.cause()));
-        return;
+    Promise<PgConnection> promise = Promise.promise();
+    if (conn != null) {
+      if (conn.failed()) {
+        promise.fail(conn.cause());
+      } else {
+        promise.complete(conn.result().conn);
       }
-      PgConnection connection = res.result();
+    } else {
+      getConnection(res -> {
+        if (res.failed()) {
+          promise.fail(res.cause());
+        } else {
+          promise.complete(res.result());
+        }
+      });
+    }
+    promise.future().onFailure(ex -> replyHandler.handle(Future.failedFuture(ex)))
+    .onSuccess(connection -> {
       String sql = SELECT + DEFAULT_JSONB_FIELD_NAME
           + FROM + schemaName + DOT + table
           + WHERE + ID_FIELD + "= $1";
+      if (lock) {
+        sql += " FOR UPDATE";
+      }
```

Comment:
```suggestion
          + WHERE + ID_FIELD + "= $1"
          + (lock ? " FOR UPDATE" : "");
```
Creating the String in one go allows the compiler to implement it using a StringBuilder.